# Miscellaneous Notes

These are loose notes with no particular theme - a kind of bookmarking.

## Markdown Tips
Markdown guides:
* [Link to Github guide](https://guides.github.com/features/mastering-markdown/)
* [A cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)

Preview or locally convert Markdown to HTML:
* [This Stack Overflow comment](http://stackoverflow.com/a/27908736) shows how.
* [Pandoc documentation](http://pandoc.org/): clearly a useful application to have.

## Julia Guides
I am still learning. Here are some Julia projects and guides:
* [Julia's home page](https://julialang.org)
* ["The Julia Express"](https://github.com/bkamins/The-Julia-Express): a guide to Julia basics
* [Tabular I/O in Julia](http://randyzwitch.com/julia-import-data/)
* [BioJulia](https://biojulia.github.io/Bio.jl/)

## Useful Software
Github Repos I've forked:
* [SISMID 2016](https://github.com/trvrb/sismid): this includes a really nice Fisher-Wright simulation. ([See here](http://stackoverflow.com/questions/19294496/wright-fisher-simulation-of-genetic-drift-using-r) for R instantiation)
* [SQL <> Pandas](https://github.com/gjreda/pydata2014nyc): cool, esp. learning how to unstack a DataFrame ([author's page](http://www.gregreda.com)).
* [ThinkStats 2nd Edition](https://github.com/AllenDowney/ThinkStats2): from Allen Downey's book.

Bioinformatics Software to Check:
* [ABRA](https://github.com/mozack/abra): Assembly Based ReAligner
* [AngularPlasmid](http://angularplasmid.vixis.com/index.php): visualisation of small circular genomes
* [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/): QC on NGS reads
* [kSNP](https://github.com/USDA-VS/ksnp): SNP detection/annotation from whole genomes
* [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/): for phylogeny ([Github repo](https://github.com/stamatak/standard-RAxML))
* [SPAdes](http://cab.spbu.ru/software/spades/): the assembler

Other Github Repos to Check:
* [bqplot](https://github.com/bloomberg/bqplot): interactive ggplot-style graphics for iPython

## Guides and notes
Bioinformatics Guides:
* [Introduction to the Data Driven Life Sciences](http://nekrut.github.io/BMMB554/): from Anton Nekrutenko (PSU)
* [RNASeq Experiment Slides](https://github.com/ucdavis-bioinformatics-training/So-you-want-to-do-a-RNAseq-experiment-Differential-Gene-Expression-Analysis/blob/master/RNAseq_Differential_Expression_Analysis.pdf)
* [An Introduction to Applied Bioinformatics](http://readiab.org)
* [SAM one-liners](https://gist.github.com/davfre/8596159)
* [IGV tutorial](https://github.com/griffithlab/rnaseq_tutorial/wiki/IGV-Tutorial)
* [RNASeq Simple Fool's Guide](http://sfg.stanford.edu/guide.html)

Organisations:
* [The Software Sustainability Institute](https://www.software.ac.uk)
* [protocol.io](https://www.protocols.io/welcome)

Folks to Keep an Eye on:
* [Anton Nekrutenko](https://galaxyproject.org): Penn State ([Github repo](https://github.com/nekrut))
* [James Taylor](http://jamestaylor.org): Johns Hopkins ([Github repo](https://github.com/jxtx))
* [David Narganes](https://github.com/davidnarganes)


Git Guides
Reference to [this link](https://vauxmaker.wordpress.com/2020/07/11/how-to-flatten-the-history-of-a-git-repository-safely/):
```
git checkout --orphan future-master
git add -A # Add all files and commit them
git commit # you will be prompted to add comment (or use -m)
git branch -D master # Deletes the master branch
git branch -m master # Rename the current branch to master
git push -f origin master # Force push master branch to github
git gc --aggressive --prune=all # remove the old files
git push --set-upstream origin master # correct the upstream branch
```